data-validation.cabal

```diff
@@ -10,34 +10,28 @@ extra-source-files:  CHANGELOG.md, README.md
 
 library
   exposed-modules:    Data.Validation
+                    , Data.Validation.Internal

```

Comment:
Why did you eliminate the internal module? It doesn't have anything to do with this issue.

Reply:
The whole point of this PR is because of issue #6

Reply:
Ah, I didn't see the update to the issue. Well, if ghcjs won't support internal modules, I guess there's nothing else we can do.

Reply:
Feels like something funky is going on here, as we definitely consume libs with multiple modules. And our exe vs lib setup is effective this as well.

Reply:
Sam is still trying to discover what is going on here, which is why this is a Draft PR. If he can't move past this, then it's a backup plan to unblock him.